include/utils/error/error_factory.hpp

```diff
@@ -52,7 +51,7 @@ class ErrorFactory {
     std::optional<std::string> default_message;
     std::optional<std::string> default_vendor_id;
 
-    const std::shared_ptr<ErrorTypeMap> error_type_map;
```

Comment:
I suppose the intention was to say a shared pointer to a const ErrorMap and not a const shared pointer to a (mutable) ErrorMap